### ISSUE TITLE
Lambda role names are not having `lambdaRole` at the end removed

### ISSUE THREAD

**Issue by j0b0sapi3n** [functionalone/serverless-iam-roles-per-function#26]
For some reason starting today when I deploy, the `-lambdaRole` at the end of the roles is no longer being removed automatically, so many of my functions are hitting the 64 character limit.

I have a feeling it might be because it's showing my environment now as NodeJS 10.10.0 because I haven't changed anything else. The weird thing is my node is set to 8.10.0 in `provider` so not exactly sure what's going on.

**Comment by meije702** [comment]
Having the same issue over here, had to use manual names to circumvent.
sls 1.45.1
this plugin: 2.0.1

**Comment by theburningmonk** [comment]
@glicht looks like `this.getRoleNameLength(fnJoin[1])` is returning `NaN`, because the latest version of SLS is returning `naming.getRoleName()` as 

```
[ 'aws-nodejs',
  'dev',
  'hello',
  { Ref: 'AWS::Region' },
  'lambdaRole' ]
```
so `{ Ref: 'AWS::Region' }` is killing your logic here

**Comment by theburningmonk** [comment]
I'd do a PR, but upgrading to the latest version of SLS seems to break the tests, I have raised a ticket in the serverless repo https://github.com/serverless/serverless/issues/6259

**Comment by jhecking** [comment]
It looks like this issue was fixed in v2.0.2 in 84e2fe6a945e2d05ee85f97364a432a96af419f1 already?

**Comment by jhecking** [comment]
The problem with the "fix" in v2.0.2 is that it simply takes the length of the `Ref` value to check the role name length, i.e. it literally just adds the length of the string `AWS::Region`. But of course the reference will be replaced with the actual region name during deployment, e.g. `ap-southeast-1` in our case. So we're still running into the issue with role names exceeding the 64 char limit for certain functions where the role name exceeds that limit but just a few characters.

Is there a way for the plugin to resolve the reference when calculating the role name length? Or can it at least determine the region name specifically?
